### PR TITLE
Basic app layout and theme

### LINKED ISSUE #1: Theme and basic layout
As a developer I need a basic layout to start building the software UI

### PR DESCRIPTION
**Pull request by Kaemppi**
resolves #1

### CODE REVIEW

File: src/components/layouts/side-panel-layout.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Paper, useMediaQuery } from "@mui/material";
+import Header from "components/layout-components/header";
+import React from "react";
+import { Content, ContentContainer, Root } from "styled/layouts/side-panel-layout";
+import theme from "theme";
+
+interface Props {
```
Comment (lauriLukkarinen) [inline src/components/layouts/side-panel-layout.tsx:7]:
```suggestion
/**
 * Component properties
 */
interface Props {
```

File: src/components/layouts/stack-layout.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import Header from "components/layout-components/header";
+import React from "react";
+import { Content, Root } from "styled/layouts/stack-layout";
+
+interface Props {
```
Comment (lauriLukkarinen) [inline src/components/layouts/stack-layout.tsx:5]:
```suggestion
/**
 * Component properties
 */
interface Props {
```

File: src/components/layout-components/navigation-item.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { ListItemText, useMediaQuery } from "@mui/material";
+import React from "react";
+import { NavigationButtonIcon, NavigationButton } from "styled/layout-components/navigation-item";
+import theme from "theme";
+
+interface Props {
+  icon?: JSX.Element;
+  title: string;
+  selected: boolean;
+  onClick(): void;
```
Comment (lauriLukkarinen) [inline src/components/layout-components/navigation-item.tsx:10]:
```suggestion
  selected?: boolean;
  onClick?: () => void;
```

File: src/components/screens/admin-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import React from "react";
+
+/**
+ * Admin screen component
+ */
+const AdminScreen: React.FC = () => {
+  /**
+   * Side navigation content
+   */
+  const renderSideNavigation = () => (
```
Comment (lauriLukkarinen) [inline src/components/screens/admin-screen.tsx:15]:
```suggestion
const AdminScreen: React.FC = () => {
  const { dropdownSettings } = strings.adminScreen.navigation;

  /**
   * Side navigation content
   */
  const renderSideNavigation = () => (
```

File: src/components/screens/admin-screen.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            onClick={ () => {} }
+          />
+          <NavigationItem
+            title={ strings.adminScreen.navigation.dropdownSettings.hazardousMaterialsAdditionalInfo }
+            selected={ false }
+            onClick={ () => {} }
+          />
+        </AccordionDetails>
+      </Accordion>
+    </List>
```
Comment (lauriLukkarinen) [inline src/components/screens/admin-screen.tsx:51]:
```suggestion
    <List>
      <Accordion defaultExpanded disableGutters>
        <AccordionSummary expandIcon={ <ExpandMore/> }>
          <Typography>
            { dropdownSettings.title }
          </Typography>
        </AccordionSummary>
        <AccordionDetails>
          <NavigationItem title={ dropdownSettings.reusableMaterials } selected/>
          <NavigationItem title={ dropdownSettings.wasteMaterials }/>
          <NavigationItem title={ dropdownSettings.postProcessing }/>
          <NavigationItem title={ dropdownSettings.hazardousMaterials }/>
          <NavigationItem title={ dropdownSettings.hazardousMaterialsAdditionalInfo }/>
        </AccordionDetails>
      </Accordion>
    </List>
```

File: src/components/screens/new-survey-screen.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+          <Button
+            size={ isMobile ? "medium" : "small" }
+            disabled={ true }
+          >
+            { strings.newSurveyScreen.createSurvey }
+          </Button>
+          <CreateManuallyButton
+            size={ isMobile ? "medium" : "small" }
+            variant="outlined"
+            onClick={ () => history.push("/survey")}
```
Comment (lauriLukkarinen) [inline src/components/screens/new-survey-screen.tsx:73]:
```suggestion
            onClick={ () => history.push("/survey") }
```

File: src/components/screens/new-survey-screen.tsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        <Stack
+          direction={{
+            xs: "column",
+            md: "row"
+          }}
+          spacing={ 2 }
+        >
+          <Button
+            size={ isMobile ? "medium" : "small" }
+            disabled={ true }
```
Comment (lauriLukkarinen) [inline src/components/screens/new-survey-screen.tsx:66]:
```suggestion
            disabled
```

File: src/components/screens/survey-screen.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        onClick={ () => {} }
+      />
+      <NavigationItem
+        icon={ <Summarize/> }
+        title={ strings.surveyScreen.navigation.summary }
+        selected={ false }
+        onClick={ () => {} }
+      />
+    </List>
+  );
```
Comment (lauriLukkarinen) [inline src/components/screens/survey-screen.tsx:73]:
```suggestion
    <List>
      <NavigationItem
        icon={ <PersonOutlined/> }
        title={ strings.surveyScreen.navigation.owner }
        selected
      />
      <NavigationItem
        icon={ <Apartment/> }
        title={ strings.surveyScreen.navigation.building }
      />
      <NavigationItem
        icon={ <NoteAdd/> }
        title={ strings.surveyScreen.navigation.others }
      />
      <NavigationItem
        icon={ <Engineering/> }
        title={ strings.surveyScreen.navigation.survey }
      />
      <NavigationItem
        icon={ <ChangeCircle/> }
        title={ strings.surveyScreen.navigation.reusables }
      />
      <NavigationItem
        icon={ <Delete/> }
        title={ strings.surveyScreen.navigation.waste }
      />
      <NavigationItem
        icon={ <WarningAmber/> }
        title={ strings.surveyScreen.navigation.hazardous }
      />
      <Divider/>
      <NavigationItem
        icon={ <Attachment/> }
        title={ strings.surveyScreen.navigation.attatchments }
      />
      <NavigationItem
        icon={ <Summarize/> }
        title={ strings.surveyScreen.navigation.summary }
      />
    </List>
  );
```

File: src/components/screens/surveys-screen.tsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+import { useHistory } from "react-router-dom";
+import { ControlsContainer, FilterRoot, NewSurveyButton, SearchBar } from "styled/screens/surveys-screen";
+import theme from "theme";
+import WhiteOutlinedInput from "../../styled/generic/inputs";
+
+/**
+ * Surveys screen component
+ */
+const SurveysScreen: React.FC = () => {
+  const [filter, setFilter] = React.useState("showAll");
```
Comment (lauriLukkarinen) [inline src/components/screens/surveys-screen.tsx:16]:
```suggestion
  const [ filter, setFilter ] = React.useState("showAll");
```

File: src/components/screens/surveys-screen.tsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+   */
+  const isMobile = useMediaQuery(theme.breakpoints.down("md"));
+
+  /**
+   * Select input handler
+   * @param event 
+   */
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFilter(event.target.value);
+  };
```
Comment (lauriLukkarinen) [inline src/components/screens/surveys-screen.tsx:30]:
```suggestion
```

File: src/components/screens/surveys-screen.tsx [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+          size={ isMobile ? "medium" : "small" }
+        />
+        <ControlsContainer direction="row" spacing={ 2 }>
+          <WhiteOutlinedInput
+            select
+            size={ isMobile ? "medium" : "small" }
+            variant="outlined"
+            id="filter"
+            value={ filter }
+            onChange={ handleChange }
```
Comment (lauriLukkarinen) [inline src/components/screens/surveys-screen.tsx:51]:
```suggestion
            onChange={ event => setFilter(event.target.value) }
```

File: src/localization/en.json [hunk 12]
```diff
@@ -1,11 +1,71 @@
...
+  "newSurveyScreen": {
+    "title": "Uusi kartoitus",
+    "description": "Hae kartoituskohdetta osoitteella, kiinteistötunnuksella tai rakennustunnuksella",
+    "address": "Address",
+    "propertyId": "Kiinteistötunnus",
+    "buildingId": "Rakennustunnus",
+    "search": "Hae",
+    "createManually": "Luo manuaalisesti",
+    "createSurvey": "Luo kartoitus"
+  },
```
Comment (lauriLukkarinen) [inline src/localization/en.json:37]: Why finnish?

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: src/localization/en.json (hunk 12 above)
Comment (Kaemppi) [reply src/localization/en.json:37]: Because hups.

File: src/components/screens/survey-screen.tsx [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+        title={ strings.surveyScreen.navigation.attatchments }
+      />
+      <NavigationItem
+        icon={ <Summarize/> }
+        title={ strings.surveyScreen.navigation.summary }
+      />
+    </List>
+  );
+
+  return (
```
Comment (lauriLukkarinen) [inline src/components/screens/survey-screen.tsx:58]:
```suggestion
  /**
   * Component render
   */
  return (
```

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**